### PR TITLE
nproc: make tests/misc/nproc-override.sh pass by implementing OMP_NUM_THREADS=X,Y,Z

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: src/uu/nproc/src/nproc.rs [hunk 1]
```diff
@@ -50,7 +50,15 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
-        Ok(threadstr) => threadstr.parse().unwrap_or(usize::MAX),
+        Ok(threadstr) => {
+            let n = threadstr.parse().unwrap_or(usize::MAX);
+            // If OMP_THREAD_LIMIT=0, rejects the value
+            if n == 0 {
+                usize::MAX
+            } else {
+                n
+            }
+        }
```
Comment (jfinkels) [inline src/uu/nproc/src/nproc.rs:61]:
```suggestion
        // If OMP_THREAD_LIMIT=0, rejects the value
        Ok(threadstr) => match threadstr.parse() {
            Ok(0) | Err(_) => usize::MAX,
            Ok(n) => n,
        }
```

File: src/uu/nproc/src/nproc.rs [hunk 2]
```diff
@@ -63,12 +71,28 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
+                } else {
+                    let n = thread[0].parse().unwrap_or_else(|_| num_cpus::get());
+                    // If OMP_NUM_THREADS=0, rejects the value
+                    if n == 0 {
+                        num_cpus::get()
+                    } else {
+                        n
+                    }
+                }
+            }
```
Comment (jfinkels) [inline src/uu/nproc/src/nproc.rs:89]:
```suggestion
            Ok(threadstr) => {
                // In some cases, OMP_NUM_THREADS can be "x,y,z"
                // In this case, only take the first one (like GNU)
                // If OMP_NUM_THREADS=0, rejects the value
                let thread: Vec<&str> = threadstr.split_terminator(',').collect();
                match &thread[..] {
                    [] => num_cpus::get(),
                    [s, ..] => match s.parse() {
                        Ok(0) | Err(_) => num_cpus::get(),
                        Ok(n) => n,
                    }
                }
            }
```

**Review by jfinkels — APPROVED** [review]
cool!

optional suggestions if we want to be slightly more concise

**Comment by sylvestre** [comment]
@jfinkels I had the feeling I could improve it but didn't find it :) many thanks !

**Pull request merged**